src/features/encoders/oxiPNG/worker/oxipngEncode.ts

```diff
@@ -77,7 +40,7 @@ export default async function encode(
   options: EncodeOptions,
 ): Promise<ArrayBuffer> {
   if (!wasmReady) {
-    wasmReady = (await threads()) ? initMT() : initST();
+    wasmReady = threads().then((hasThreads: boolean) => hasThreads ? initMT() : initST());
```

Comment:
Note: this is unrelated fix, just noticed that we weren't assiging Promise immediately, which could result in bugs.